src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -114,4 +128,71 @@ private String getUploadedFileUrl(HttpServletRequest request, String formInputEl
     } 
     return url;
   }
+    
+  /**
+   * Detects words from a picture and returns them
+   * TODO: Clean up the result, sometimes the api misreads, misses entirely, or adds additional words
+   *        
+   */
+  private static String detectDocumentText(String path) throws IOException {
```

Comment:
Would it be a good idea to consider moving this to a separate class as a static function? Not sure how else we could organize this code, but was thinking in terms of keeping things organized and readability.

I would imagine this file getting quite large if @joymaneetk were to add the NLP processing logic in this servlet as well. Does anyone else have thoughts on this? I'm okay with leaving it here as well.

Reply:
We could probably move it to a separate class holding all the OCR functions and similar with NLP functions.